agent/rpc/middleware/interceptors.go

```diff
@@ -77,6 +123,30 @@ func requestType(req interface{}) string {
 	return "unreported"
 }
 
+func (r *RequestRecorder) getServerRole() string {
+	if r.serverIsLeader != nil {
+		if r.serverIsLeader() {
+			return "leader"
+		} else {
+			return "follower"
+		}
+	}
+
+	// This logical branch should not happen. If it happens
+	// it means that we have not plumbed down a way to verify
+	// whether the server handling the request was a leader or not
+	return "unreported"
+}
```

Comment:
Actually the node can have multiple states. `IsLeader` will report true if the node is a leader. If it report false it mean the node have one of the following states:
- Follower
- Candidate
- Shutdown

Reply:
Hm, wondering if we could change the label around `IsLeader` then instead:

```
{
"is_leader": {true | false}
}
```

Reply:
reworked this func for a tri-state